adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/ClientCreds/JsonWebToken.cs

```diff
@@ -209,18 +209,19 @@ internal class JWTPayload
         [DataContract]
         internal sealed class JWTHeaderWithCertificate : JWTHeader
         {
-            private string _thumbPrint;
-
-            public JWTHeaderWithCertificate(IClientAssertionCertificate credential)
+            public JWTHeaderWithCertificate(IClientAssertionCertificate credential, bool sendX5C)
                 : base(credential)
             {
-                _thumbPrint = this.Credential.Thumbprint;
+                X509CertificateThumbprint = this.Credential.Thumbprint;
                 X509CertificatePublicCertValue = null;
 
+                if (!sendX5C)
+                    return;
+
```

Comment:
are you sure that you want to return here ?
As I understand the idea of next  check was to force usage of  library provided implementation of IClientAssertionCertificate

Reply:
looks like with that change it will be possible to use developer provided impl of IClientAssertionCertificate ,  it is api change

Reply:
@trwalke thoughts on this one?

Reply:
yes, we return here because if there is no request to send the x5c, there is no reason to check for developer implementation or set the X509CertificatePublicCertValue. The only reason we are checking for the developer implementation is because we cannot send the x5c if the developer implemented their own IClientAssertionCertificate .